File: RobotBehavior/Arbiter/ArbiterGroup.py

from RobotBehavior.Arbiter.Arbiter import Arbiter
from copy import deepcopy

class ArbiterGroup(Arbiter):
	def __init__(self, id):
		self._id = id
		self._arbiters = list()


	def process(self):
		pass

	def add_arbiter(self, arbiter):
		self._arbiters.extend(arbiter)


	def getArbiters(self):
		return deepcopy(self._arbiters)

	
"""
public  ArrayList<ArbiterConnection> getAllOutputs()
{
	ArrayList<ArbiterConnection> list = new ArrayList<ArbiterConnection>();
	return(list);
}
"""

File: RobotBehavior/Arbiter/ArbiterGroup.java
package de.hska.lat.behavior.arbiter;

import java.util.ArrayList;
import java.util.ListIterator;

public class ArbiterGroup extends Arbiter
{

	
	protected ArbiterGroup(int id)
	{
		super(id);
	}


	protected ArrayList<Arbiter>  arbiters = new ArrayList<Arbiter>(); 
	
	
	@Override
	protected void process()
	{
	}


	
	
	
public void addArbiter(Arbiter arbier)
{
	this.arbiters.add(arbier);
}
	

public  ArrayList<ArbiterConnection> getAllOutputs()
{
	ArrayList<ArbiterConnection> list = new ArrayList<ArbiterConnection>();
	return(list);
}





public ListIterator<Arbiter> getArbiters()
{
	return (this.arbiters.listIterator());
}








}
